GroceriesApp/GroceryappDesign.sql

```diff
@@ -0,0 +1,158 @@
+    CONSTRAINT UQ_UserDetails_RegisterEmail UNIQUE (Email)
+);
+
+
+
+INSERT INTO UserDetails(
+    FirstName, LastName, Email, [Password], ConfirmPassword, PhoneNumber,[Address], City, [State], PinCode, UserType
+) VALUES
+    ('John', 'Doe', 'john.doe@example.com', 'password123', 'password123', '+1234567890', '123 Main St', 'New York', 'NY', '10001', 'Customer'),
+    ('Jane', 'Smith', 'jane.smith@example.com', 'password123', 'password123', '+0987654321', '456 Maple Ave', 'Los Angeles', 'CA', '90001', 'Customer'),
+    ('Supplier One', 'Supplier', 'supplier1@example.com', 'password123', 'password123', '+9876543210', '789 Oak St', 'Chicago', 'IL', '60601', 'Supplier'),
+    ('Supplier Two', 'Supplier', 'supplier2@example.com', 'password123', 'password123', '+0123456789', '321 Pine Ave', 'Miami', 'FL', '33101', 'Supplier');
+	
+-- Verify the data insertion
+SELECT * FROM UserDetails;
+
+
+CREATE TABLE Products_Supplier (
+    ProductSupplierId INT IDENTITY(1,1),
+    ProductName VARCHAR(20) NOT NULL,
+    ProductPrice MONEY NOT NULL,
+    ProductDescription VARCHAR(50) NOT NULL,
+    ProductQuantity INT NOT NULL,
+   CONSTRAINT PK_Products_Supplier_ProductSupplierId PRIMARY KEY(ProductSupplierId),	
+);
+
+
+-- Insert dummy data into Products_Supplier
+INSERT INTO Products_Supplier ( ProductName, ProductPrice, ProductDescription, ProductQuantity)
+VALUES 
+( 'Apple', 0.50, 'Fresh Red Apple', 100),
+( 'Banana', 0.30, 'Fresh Yellow Banana', 150),
+( 'Carrot', 0.20, 'Organic Carrot', 200),
+( 'Tomato', 0.40, 'Fresh Red Tomato', 180),
+( 'Cheese', 2.00, 'Cheddar Cheese', 30);
+
+
+
+-- Verify the data insertion
+SELECT * FROM Products_Supplier;
+
+
+
+	CREATE TABLE Categories(
+    CategoryId INT  IDENTITY(1,1),
+	CategoryName VARCHAR(20)  NOT NULL,
+	CONSTRAINT UQ_Categories_CategoryName UNIQUE (CategoryName),
+	CONSTRAINT PK_Categories_CategoryId  PRIMARY KEY(CategoryId),
+
+);
+
+
+INSERT INTO Categories( CategoryName) VALUES
+    ( 'Fruits'),
+    ( 'Vegetables'),
+    ( 'Dairy');
+
+	
+-- Verify the data insertion
+SELECT * FROM Categories;
+
+CREATE TABLE Products(
+ ProductsId INT IDENTITY(1,1),
+ ProductSupplierId INT,
+ ProductName VARCHAR(20) NOT NULL,
+ ProductPrice MONEY NOT NULL,
+ ProductDescription VARCHAR(50) NOT NULL,
+ ProductQuantity INT NOT NULL,
+ CategoryId INT,
+ CONSTRAINT FK_Products_Catergories FOREIGN KEY(CategoryId) REFERENCES Categories(CategoryId),
+ CONSTRAINT PK_Products_ProductId PRIMARY KEY(ProductsId),
+ CONSTRAINT FK_Products_ProductSupplierId FOREIGN KEY(ProductSupplierId) REFERENCES Products_Supplier(ProductSupplierId)
+);
+
+
+
+INSERT INTO Products (ProductSupplierId, ProductName, ProductPrice, ProductDescription, ProductQuantity, CategoryId)
+VALUES
+(1, 'Apple', 0.50, 'Fresh Red Apple', 100, 1),
+(2, 'Carrot', 0.20, 'Organic Carrot', 200, 2),
+(5, 'Cheese', 2.00, 'Cheddar Cheese', 30, 3);
+
+-- Verify the data insertion
+SELECT * FROM Products;
+
+
+
+CREATE TABLE [Orders](
```

Comment:
How are you differentiating this order belongs to which customer and what they bought?